### PR TITLE
Modify scale-down-ignore-pdb flag to not affect pods controlled by StatefuSets.

### PR DESCRIPTION
**Pull request by jodzga**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/simulator/drain.go [hunk 1]
```diff
@@ -55,7 +56,10 @@ func FastGetPodsToMove(nodeInfo *schedulerframework.NodeInfo, skipNodesWithSyste
 	}
 	// if scaleDownIgnorePDB is true, we ignore PDBs when checking if a pod is blocking
 	if scaleDownIgnorePDB {
-		return pods, daemonSetPods, nil, nil
+		// if there are no statefulset pods, we can ignore PDBs
+		if !podsIncludeStatefSet(pods) {
```
Comment (leekathy) [inline cluster-autoscaler/simulator/drain.go]: typo? `podsIncludeStatefulSet`

File: cluster-autoscaler/simulator/drain.go [hunk 2]
```diff
@@ -89,7 +104,10 @@ func DetailedGetPodsForMove(nodeInfo *schedulerframework.NodeInfo, skipNodesWith
...
```
Comment (leekathy) [inline cluster-autoscaler/simulator/drain.go]: typo ditto

File: cluster-autoscaler/simulator/drain.go [hunk 3]
```diff
@@ -64,6 +68,17 @@ func FastGetPodsToMove(nodeInfo *schedulerframework.NodeInfo, skipNodesWithSyste
...
+func podsIncludeStatefSet(pods []*apiv1.Pod) bool {
+	podsIncludeStatefSet := false
+	for _, pod := range pods {
+		if pod_util.IsStatefulSetPod(pod) {
+			podsIncludeStatefSet = true
+			break
+		}
+	}
+	return podsIncludeStatefSet
+}
```
Comment (leekathy) [inline cluster-autoscaler/simulator/drain.go]: typo ditto

File: cluster-autoscaler/main.go [hunk 4]
```diff
@@ -183,7 +183,7 @@ var (
 
 	emitPerNodeGroupMetrics = flag.Bool("emit-per-nodegroup-metrics", false, "If true, emit per node group metrics.")
 
-	scaleDownIgnorePDB = flag.Bool("scale-down-ignore-pdb", false, "If true, cluster-autoscaler will ignore PDBs and may evict pods violating PDBs during scale-down.")
+	scaleDownIgnorePDB = flag.Bool("scale-down-ignore-pdb", false, "If true, cluster-autoscaler will ignore PDBs and may evict pods violating PDBs during scale-down by directly deleting pods. This flag does not affect pods controlled by StatefulSets.")
```
Comment (leekathy) [inline cluster-autoscaler/main.go]: IMO from the second sentence, it's not super clear which of the following behaviors is taking place:
- CA does not downscale nodes with statefulset pods even if their PDBs can be respected
- CA does not downscale nodes with statefulset pods if any of their PDBs will be violated
- CA does not directly delete statefulset pods when scaling down nodes

**Review by leekathy — APPROVED** [review]
cool! just a few nits

File: cluster-autoscaler/main.go (hunk 4 above)
Comment (jodzga) [reply cluster-autoscaler/main.go]: Thanks for the feedback! Will rephrase.

File: cluster-autoscaler/utils/pod/pod.go [hunk 5]
```diff
@@ -28,6 +28,16 @@ const (
 	DaemonSetPodAnnotationKey = "cluster-autoscaler.kubernetes.io/daemonset-pod"
 )
 
+// IsDaemonSetPod returns true if the Pod should be considered as Pod managed by a DaemonSet
```
Comment (yanbinren) [inline cluster-autoscaler/utils/pod/pod.go]:
```suggestion
// IsStatefulSetPod returns true if the Pod should be considered as Pod managed by a StatefulSet
```

**Review by yanbinren — APPROVED** [review]

File: cluster-autoscaler/utils/pod/pod.go (hunk 5 above)
Comment (jodzga) [reply cluster-autoscaler/utils/pod/pod.go]: Fixed

File: cluster-autoscaler/main.go (hunk 4 above)
Comment (jodzga) [reply cluster-autoscaler/main.go]: Rephrased

**Pull request merged**